clang/include/clang/AST/DeclTemplate.h

```diff
@@ -860,6 +860,16 @@ class RedeclarableTemplateDecl : public TemplateDecl,
   /// \endcode
   bool isMemberSpecialization() const { return Common.getInt(); }
 
+  /// Determines whether any redeclaration of this template was
+  /// a specialization of a member template.
+  bool hasMemberSpecialization() const {
+    for (const auto *D : redecls()) {
+      if (D->isMemberSpecialization())
+        return true;
+    }
+    return false;
```

Comment:
At least for RedeclarableTemplate, there is a common area which is shared by all redeclarations of a decl.
This could be implemented as a bit there, which would make sense in case the common area is used, thus allocated, for these users anyway.

Not required, just pointing out the possibility.

Reply:
I've actually been investigating implementing a "common data" structure shared by all redeclarations of `Redeclarable` derived types for a while now (at least since last year). For now, I'd like to just merge a patch that works. In the future. I plan to further enhance this implementation once I unify how we store data common to all redeclarations.